sources/editor/Stride.Assets.Presentation/AssetEditors/GameEditor/ViewModels/EditorGridViewModel.cs

```diff
@@ -50,19 +50,67 @@ public EditorGridViewModel([NotNull] IViewModelServiceProvider serviceProvider,
         /// </summary>
         public int AxisIndex { get { return Service.AxisIndex; } set { SetValue(AxisIndex != value, () => Service.AxisIndex = value); } }
 
+        /// <summary>
+        /// Gets or sets whether the X axis of the grid is currently visible.
+        /// </summary>
+        public bool IsXAxisVisible { get => Convert.ToBoolean(AxisSelection & 0b001); set => AxisSelection = Convert.ToInt16(value) ^ (AxisSelection & 0b110); }
+
+        /// <summary>
+        /// Gets or sets whether the Y axis of the grid is currently visible.
+        /// </summary>
+        public bool IsYAxisVisible { get => Convert.ToBoolean(AxisSelection & 0b010); set => AxisSelection = (Convert.ToInt16(value) << 1) ^ (AxisSelection & 0b101); }
+
+        /// <summary>
+        /// Gets or sets whether the Z axis of the grid is currently visible.
+        /// </summary>
+        public bool IsZAxisVisible { get => Convert.ToBoolean(AxisSelection & 0b100); set => AxisSelection = (Convert.ToInt16(value) << 2) ^ (AxisSelection & 0b011); }
+
+        private int AxisSelection { get => ConvertAxisIndexToSelection(AxisIndex); set => AxisIndex = ConvertAxisSelectionToIndex(value); }
+
         /// <summary>
         /// Gets a command that will toggle the visibility of the grid.
         /// </summary>
         public ICommandBase ToggleCommand { get; }
 
         private IEditorGameGridViewModelService Service => controller.GetService<IEditorGameGridViewModelService>();
 
+        private int ConvertAxisIndexToSelection(int index)
+        {
+            switch (index)
+            {
+                case 0: return 0b001;
+                case 1: return 0b010;
+                case 2: return 0b100;
+                case 3: return 0b011;
+                case 4: return 0b101;
+                case 5: return 0b110;
+                case 6: return 0b111;
+            }
+            return 0b000;
+        }
+
+        private int ConvertAxisSelectionToIndex(int selection)
+        {
+            switch (selection)
+            {
+                case 0b001: return 0;
+                case 0b010: return 1;
+                case 0b100: return 2;
+                case 0b011: return 3;
+                case 0b101: return 4;
+                case 0b110: return 5;
+                case 0b111: return 6;
+            }
+            return 7;
+        }
+
         public void LoadSettings([NotNull] SceneSettingsData sceneSettings)
         {
             IsVisible = sceneSettings.GridVisible;
             Color = sceneSettings.GridColor;
             Alpha = sceneSettings.GridOpacity;
             AxisIndex = sceneSettings.GridAxisIndex;
+            AxisSelection = ConvertAxisIndexToSelection(sceneSettings.GridAxisIndex);
```

Comment:
Is the `AxisIndex` used for anything else? If not, then I would suggest replacing it completely with the `AxisSelection` and passing that to the ViewportGridGizmo. Moreover, the bit mask can be encoded in an enum:
```
[Flags]
public enum ViewportGridAxis
{
    X = 1,
    Y = 2,
    Z = 4,
}
```
Which allows you to do combined enum values like `ViewportGridAxis.X & ViewportGridAxis.Y`.

Reply:
You are right. Thanks for your recommendation!